hail/src/main/scala/is/hail/expr/ir/EmitClassBuilder.scala

```diff
@@ -472,21 +476,19 @@ class EmitClassBuilder[C](
   }
 
   def getPType[T <: PType : TypeInfo](t: T): Code[T] = {
-    val references = ReferenceGenome.getReferences(t.virtualType).toArray
-    val setup = Code(Code(references.map(addReferenceGenome)),
-      Code.checkcast[T](
+    ReferenceGenome.getReferences(t.virtualType).toArray.map(getReferenceGenome)
```

Comment:
so I didn't realize this earlier, but the reason these methods serialize to strings and parse is that they don't want to serialize a RG inside a locus type and end up with multiple distinct RG objects for the same reference on the worker. That's not a problem anymore! You can just use `mb.getObject(t)` for both these implementations! 🎉